### PR TITLE
Add tests for pyeight.eight and pyeight.user

### PR DESCRIPTION
**Pull request by raman325**
I made minimal changes here to get pylint to pass. Next step after merge would be to simplify some of the logic, to add type hints, and to improve coverage

### CODE REVIEW

File: pyeight/user.py [hunk 1]
```diff
@@ -10,14 +10,15 @@
 from datetime import datetime, timedelta
 import logging
 import statistics
-import time
+
+import pytz
```
Comment (mezz64) [inline pyeight/user.py:14]: Can you add this functionality without the pytz library?  I'd like to stay away from bundling in non-standard libraries if possible.

Comment (raman325) [reply pyeight/user.py:14]: as far as I can tell, not unless we set the minimum version of the package to 3.9 which introduced this: https://docs.python.org/3/library/zoneinfo.html

Comment (raman325) [reply pyeight/user.py:14]: FWIW the latest version of HA sets 3.9 to a minimum, so if we are primarily concerned with HA, we can update the minimum. I'm assuming old versions of the package are still available for older Python versions

Comment (raman325) [reply pyeight/user.py]: I went ahead and made that change - if you'd like me to revert the last commit just let me know, but this:
- Bumps min version to 3.9
- uses stdlib zoneinfo for timezones

**Comment by raman325** [comment]
I also added another round of tests to improve coverage and this is my final list for now. This will help ensure that when I make a bunch of changes to the code (already ready and I am testing it against these tests), the test case changes should be minimal and deliberate in that I think the behavior should be what the new behavior will be

**Comment by raman325** [comment]
good call, that was meant to be a placeholder and I certainly don't want my email in there 🙂

File: tests/test_eight.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+"""Tests for the eight module."""
+from pyeight.eight import EightSleep
+
+
+async def test_setup(client_session):
+    """Test the setup of the eight module."""
+    eight = EightSleep(
+        "raman325@gmail.com", "password", "UTC", client_session=client_session
```
Comment (mezz64) [inline tests/test_eight.py:8]: Email update.

File: tests/test_eight.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    await eight.stop()
+    # Assert that the client session is not closed since we created it outside the lib
+    assert not client_session.closed
+
+
+async def test_update_device_data(client_session):
+    """Test update device data."""
+    eight = EightSleep(
+        "raman325@gmail.com", "password", "UTC", client_session=client_session
```
Comment (mezz64) [inline tests/test_eight.py:26]: Email update.

File: tests/test_user.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    await eight.update_device_data()
+    assert user.past_heating_level(1) == -32
+
+    await eight.stop()
+
+
+async def test_update_user_data_missing_data(client_session_missing_data):
+    """Test update user data with missing data."""
+    eight = EightSleep(
+        "raman325@gmail.com",
```
Comment (mezz64) [inline tests/test_user.py:120]: Email update.

**Comment by mezz64** [comment]
> good call, that was meant to be a placeholder and I certainly don't want my email in there 🙂

I figured.  There are still a few more instances.  I marked them in review so you can find them easily.

**Pull request merged**